### ISSUE TITLE
🔃 🚀 Feature Request: Make Elementor fully GDPR compliant

### ISSUE THREAD

**Issue by IDESIGNSTUDIO** [elementor/elementor#4544]
Elementor itself might collect data if we agree and activate this option on, so on this level it is GDPR compliant.

But on the other hand, Elementor is using third party services like Google Maps, Google Web Fonts, Social Share buttons, Facebook API, reCaptcha etc.

Each of these services collect user sensitive data.

Elementor must develop further to comply with using those with GDPR. Currently, I am exchanging many of the in-built features of Elementor with similar but GDPR compliant, as for example Shariff for secure GDPR compliant social share. Double work, double scripts, for something which is basically already there.

This compliance must be taken care of, otherwise by using some of the in-built features from Elementor we are creating websites for clients that are actually breaking the EU Privacy Law. Very dangerous!

Please have a look on this request with care and the point of view of the most of Elementor users.
Thank you!

**Comment by IDESIGNSTUDIO** [comment]
There is already a plugin in the repository "Extra privacy for Elementor". May be the dev team should have a look on it?
https://wordpress.org/plugins/extra-privacy-for-elementor/

**Comment by bainternet** [comment]
@IDESIGNSTUDIO 

Can you explain what you mean by:
 
> Elementor must develop further to comply with using those with GDPR

and how is elementor social share widget not GDPR complaint?

**Comment by deckerweb** [comment]
@IDESIGNSTUDIO 
GDPR compliant usage of Google Maps and Google Fonts is only possible by declaring this stuff in your privacy policy and making contract with Google. Elementor can do nothing about that it's your turn.
But: Elementor Pro already has custom fonts and you can upload Google Fonts there to let them load locally - one issue solved.

And for Facebook API and reCaptcha: just don't use these widgets as of now. What should Elementor do to make those compliant? Cannot imagine. These are already third-party services, you have to care for yourselve when using those services and check between you and these services.

@bainternet 
Sharing Buttons:
As long as Elementor is only using the "classic" sharing functions of these services then no data is transferred (like the IP address etc.). But you should declare in public what data is transferred if at all, so anyone of us can decide what to do. But: if any data is transferred by default with these sharing buttons then please add an option to anonymize all those data or an option to have no data transferred but only the "classic" sharing used.

I see that for the counters a third-party service called "donReach" is used: when this service is not transferring any personal data like IP or similar then it is ok and Elementor might be compliant in that regard.
However, you as the admin of the site have to care how "donReach" is handling those counter data and all that goes along with it. This is not a task of Elementor.

**Comment by deckerweb** [comment]
Looks, like I am wrong with my above opinion regarding Google Maps usage in regards of GDPR:
I just stumpled about this plugin https://codecanyon.net/item/advanced-google-maps-plugin-for-wordpress/5211638 by accident and they added GDPR compliance and wrote this (scroll down on page to find statement):
"This plugin will loads google maps API & google maps once visitor will accept the cookies policy, privacy policy & term & condition. No scripts or maps will be loaded until your visitors’ consent cookies."

That would mean Elementor should add similar settings that can optionally be used for those who need GDPR compliance; adding the consent saving to WP's GDPR stuff would also make sense (things added with WP 4.9.6)

However, since Google Maps will change things on June 11th 2018 for usage (add credit card in dev console etc.) I would love Elementor pursuing other alternatives additionally to Google Maps, like OpenStreetMap or Bing Maps. Also with GDPR compliance if needed, of course...

Also, there is already this third-party plugin regarding Elementor privacy/ GDPR:
https://wordpress.org/plugins/extra-privacy-for-elementor/
--> it helps already for YouTube Videos and Google Maps embedding...!

**Comment by IDESIGNSTUDIO** [comment]
Thank you all for taking part in the discussion. GDPR is an important topic, since many EU companies use Elementor to build websites for their customers, or company websites from other countries that try to do business with EU citizens. 

I am just aiming to motivate Elementor's team and looking for their help to implement bulletproof GDPR technical features for the third party services, which we can easily use when designing a website that must comply with GDPR.

So, the suggestion to Elementor's team and community is instead of trying to install GDPR compliant plugins for Social Share, Google Maps, etc. to simply work in direction to make the in-built features in Elementor _technically_ GDPR compliant. 

Here are some of them:

1. **Social Share** in a way that there is no user privacy data tracking involved (would "classic" sharing functions be the solution ? ) - then there will be no need to install Shariff Wrapper plugin for example and use what Elementor already offers. 

2. **Google Maps** Opt-in / consent to give. Opt-Out is not enough. The visitor must explicitly agree on using Google Maps. Only after agreement the Google Maps (scripts and codes) should be loaded. Can we use as an example to follow the Codecanyon plugin? 
Again then, we as EU-websites designers would not need to install another plugin for Google Maps that is GDPR compliant, but use what Elementor offers as in-built functionality.

3. **YouTube** could be similar as P.3. To be on the secure side for now and any future changes that might come from Google regarding their Privacy Policy, also a request of a consent before any video scripts are loaded to be added as an option?
Could a checkbox consent as a video placeholder or video overlay might be a solution?

4. With **Google Fonts** we have the solution as @deckerweb described it. Great!

5. What about **reCaptcha**? 
reCaptcha v2 is registering a cookie on the computer only after the checkbox 'I am not a robot' is clicked. But the reCaptcha scripts themselves are already loaded on page load without user's agreement. Regarding Google's Privacy Policy we must notify the visitor and ask for their consent to use reCaptcha.

The way I understand it and think of it, is that it is getting complicated. The visitor must agree of allowing reCaptcha, so then the reCaptcha loads on the website, so the user can click on the button "I am not a robot".

Does this make any usability sense !?

Would it be a solution to have an Anti-Spam field in Elementor's contact form fields? Anything like a simple math or to submit answer to a question. It is not so cool as reCaptcha, but if it keeps the reCaptcha Privacy Policy bother away and if it protects against spam, then it sounds good to me.

6. For any other 3rd party service where the Privacy Policy from the service provider is really clear with IP anonymisation, or no IP collection, no further distribution of data collected, etc., we have to state all that on the website Privacy Policy page, and also in the Privacy Center (where Cookies are listed and described).

7. About MailChimp API & usage, and other integrations I have to look at it with testing still. Anyone having a bullet proof way to GDPR comply for those?

Please share any observations and knowledge. At the end of the day we will solve questions and technical issues that many web designers or companies using Elementor must deal with. 

I am absolutely ready to support Elementor's team by paying for an extra extension for GDPR compliance of third party services provided as features in this great plugin.

Thanks again for the great share.

**Comment by themedleb** [comment]
**## The perfect and complete GDPR cookie consent for me is to:**

**1.** Give the site visitor the ability to turn off additional services from the cookie consent interface using switch buttons, and if the experience looks stacked and cluttered these services should be organized in tabs.

**2.** Be multilingual (Like WPML support).

**3.** Support the most popular plugins and services like form builder plugins/services (Gravity Forms, Ninja Forms, ...) statistics/analytics plugins/services (Google Analytics, ...) email marketing plugins/services (MailChimp, ...), social media plugins/services.

**4.** Have a shortcode to paste in a page that is visible to every user (Registered or Non-registered) where all these options will be there saved (If chosen/configured before), pretty useful if a user wants to turn on/off a service later, so the user will always have control of the offered services even after the first cookie consent commitment.

I hope this helps.

**Comment by ChriKn** [comment]
Funny this still hasn't been addressed, we shouldn't have to install yet another plugin to get some basic (and horrendously idiotic) legal stuff to work...

**Comment by nickcj931** [comment]
Has there been any news on this?

**Comment by nesoor** [comment]
I completely agree with this !
Also the youtube embed isn't GDPR compliant so there is still a lot of work for Elementor.

**Comment by pgabrielandrei** [comment]
Regarding the Google Maps issue, could the Elementor devs not solve this by using the Google Maps **javascript** API, instead of the **Embed** API, which relies on **iFrames**? I found this on a forum:

> The Google Maps JavaScript API creates a map using JavaScript. There are no AJAX requests being sent to Google or anywhere else that contains identifiable information (or any information for that matter).
> 
> The only request that is sent to Google when loading a map is the request to download the JS file. You can confirm this by inspecting the Network tab of the Google Chrome developer tools. The request to pull the JS file does not contain any information from the user and there are no AJAX requests that follow that call.

**Comment by Micsi** [comment]
I second the request for GDPR compliance for widgets in Elementor (Pro at least). My concern is with the social share widgets and google maps integration. Help for getting google fonts local would be much appreciated, though, but that's probably more of an issue for the theme used.

I don't think there is an issue with using recaptcha. GDPR allows you to collect data if you have good reason for. Recaptcha is a security feature to protect the site and data stored within so that's a very legal reason to collect an use data. You have to declare the usage in your privacy policy, that's it.

At the moment, I am looking for a good solution regarding share buttons. I would really love to see a widget integrating Shariff. Shariff is the best solution I've come across. I uses a two click solution. The first click loads the necessary Facebook libraries or whatever service you want to share on, the second does the actual sharing.

**Comment by IDESIGNSTUDIO** [comment]
I am curious if anyone from Elementor is reading and following up on this post. There are no comments on this feature request from their side, even if the community here gave a lot of hits of right, wrong and necessary.

**Here is an example of Avada theme**. 
If Google Maps must be loaded on a page, there is first a placeholder with the message that GMaps will collect in this case some privacy related info. There is an agree button and only when the user clicks it, the maps loads.
I find this solution extremely elegant.
Avada is perfectly handling Google Fonts,, YouTube etc.

Another very popular WP theme - **Enfold**, also has in-built functionality to handle GDPR compliance. Also very clever and beautifully made.

If Elementor is **targeting web designers to design a beautiful website without coding**, then please, **don't leave in their non-programmers hands to deal things like GDPR, javascript and cookies**!!!

I am actually looking already for other solutions than Elementor, so we can pull projects faster than currently just because of the GDPR compliance. It takes more time to integrate 3rd party solution.  There is the need to install plugins with similar functionality as Elementor, just because they are doing the GDPR compliance RIGHT. All this is more scripts, more time to load, more time to optimize a website. At the end stops to be as cost effective as it is with better all-rounded solutions available.

If anyone has similar concerns, please put your voice here, so this future request could be considered from the devs. Elementor is loosing big on this.

**Comment by wpglossary** [comment]
> If anyone has similar concerns, please put your voice here, so this future request could be considered from the devs.

I absolutely second this! GDPR forces anyone designing new solutions in any area, not only websites, to practice privacy by design. This unfortunately absolutely disqualifies Elementor as a tool to design web sites as you need to work around it when it comes to GDPR.

To create a site that is GDPR compliant in Elementor is much unneeded work and pretty complicated, although Elementor could (and should) be a big help in creating a compliant site. I think, the team is missing on a big opportunity here.

As a certified datasecurity officer here in germany I am pretty interested in this and willing to help

**Comment by mgustavs** [comment]
I ran into one more addition to the list - Elementor Pro automatically generated localStorage/sessionStorage (#7151) with no in-plugin option to stop it.

It would be lovely (and reasonable I think) to see a cookie consent pop-up from Elementor, which not only looks nice but also solves the internal storage and other cookie and other privacy-related issues by actively blocking them prior consent.

**Comment by yasuoiwakura** [comment]
Why is this not solved yet?
google maps, google fonts, fontawesome - i dont even use maps but the visitors IP is sent to google! wtf?
i startet building pages with elementor in 2019 but if this matter is not taken seriously it becomes irresponsible using it further. Makes no sense to use a time-saving page-builder but waste your time to workaround gdpr-flaws :-/

**Comment by ianstudio** [comment]
+1

**Comment by RomkeJellema** [comment]
+1

**Comment by ellegaarddk** [comment]
I agree with this. We NEED this GDPR compliancy!

**Comment by centralplexus** [comment]
Agreed. This is long overdue now and ultimately not at all optional. It's time to move forward before compliance forces us to utilize a different page builder.

**Comment by Larzans** [comment]
+1

**Comment by AertHulsebos** [comment]
Hi,

Please refer to this issue; https://github.com/elementor/elementor/issues/10659, there's a Consent API coming in WordPress core. 

Also: https://wptavern.com/proposal-to-add-a-consent-api-to-wordpress-feature-plugin-available

Will be likely to feature in the last core update of 2020.

regards Aert

**Comment by sybilrondeau** [comment]
HI, 

Does anyone know what is the elementor statistic cookie placed ? I did not check the box in Elementor settings... 

Regards

**Comment by sybilrondeau** [comment]
What's more, have you seen that : https://cookiedatabase.org/service/elementor/ ? 

That means Elementor isn't totaly GDPR compliant...

**Comment by Virginie-D** [comment]
Hi there, 
thanks for this very interesting post.

Even if there is already an answer given for the Google Font issue, I wonder : 
Is the Custom Fonts for Elementor enough to be sure there is no other call or request to the google library ? 

is there any benefits to add the OMGF plugin ?
https://wordpress.org/plugins/host-webfonts-local/

In Elementor >> Settings >> Advanced >> google Fonts Load :  which method is the best when you have upload your own fonts ? 

Thanks for your answer regarding to your experience

**Comment by sybilrondeau** [comment]
Hi @Virginie-D, For me it's enough. If you check with the network panel in Chrome Dev Tools there is no more call to Google API. 

And I was wrong about Elementor not compliant : It is. The cookie I was talking about is local.

**Comment by Virginie-D** [comment]
> Hi @Virginie-D, For me it's enough. If you check with the network panel in Chrome Dev Tools there is no more call to Google API.
> 
> And I was wrong about Elementor not compliant : It is. The cookie I was talking about is local.

Thanks Sybil !

**Comment by Larzans** [comment]
The GDPR also forbids you from requesting any file from a third party IP without asking the user first, but there is a hard coded call to pre-fetch google files, so that is NOT GDPR compliant.

**Comment by m1ga** [comment]
Hey @clmnsm, try to switch on `Inline Font Icons Experiment` :
> The “Inline Font Icons” will render the icons as inline SVG without loading the Font-Awesome and the eicons libraries and its related CSS files and fonts. `

**Comment by clmnsm** [comment]
This doesn't seem to do anything on latest Elementor/Pro 3.8.0

Both the icons in buttons and the icon list element still use font based icons even after enabling the experiment and refreshing CSS.

**Comment by m1ga** [comment]
should use the svg:
![Screenshot_20221101_145803](https://user-images.githubusercontent.com/4334997/199250739-67175845-dae7-445f-a863-4009bf90f844.png)

**Comment by clmnsm** [comment]
Inline font icons only work when no FA kit id is supplied and when you use the basic (non-pro) Font Awesome icons.

So it seems like it doesn't work with the pro fonts like light or duotone yet. 

So it's not an option for existing projects that already use lots of pro icons.

**Comment by m1ga** [comment]
I see. But the page https://elementor.com/help/inline-font-awesome/?utm_source=experiments&utm_medium=wp-dash&utm_campaign=learn has some info about that too:
> Note: This feature does not convert Font Awesome Pro icons. If you are a Pro FA user, navigate to your Kit settings at the Font Awesome site at Kit > Settings > Technology and set them to SVG.

Not using it but sounds worth a try

**Comment by clmnsm** [comment]
Interesting find - this does change them to SVG, but the colors are always default (the CSS rule erroneously uses "fill" instead of "color" in the icon list element) and the SVGs themselves are still loaded from the remote server.

So no luck with getting Font Awesome to be GDPR compliant yet.

**Comment by sebastiantheobald** [comment]
I run into the same problem like @clmnsm - their has to be a solution to use alle the font awesome icons (pro) through the elementor icon library in a gdpr compliant way. Very urgent!

**Comment by toineenzo** [comment]
This just got merged

https://github.com/elementor/elementor/pull/29732

**Comment by clmnsm** [comment]
> This just got merged
> 
> #29732

This doesn't seem to affect font awesome, as far as I can see. 

Local google fonts are nice, but have been possible via OMGF for a long time. Not having to install another plugin is great though.

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that Local Google fonts are now a native feature of Elementor since the release of version 3.27.0! 🥳 

You can now activate the new experimental feature by going to your WordPress **`Dashboard`** > **`Elementor`** > **`Settings`** > **`Features`** page, and switching the option to **`Load Google Fonts locally`** to **`Active`**.

This will make Elementor automatically download all the weights and style files of the Google Fonts in use in the **`.woff2`**, and save them in a special folder inside the **`../wp-content/uploads/`** folder.

These font files are downloaded only once a page using the Google Fonts is loaded, and will be re-downloaded on every new visit after the **`Elementor CSS & Data`** is **`Regenerated`**. Fonts are only loaded when in use.

For more information see: https://elementor.com/help/load-google-fonts-locally/

In the future this feature will be merged to core and may or may not have a setting in the **`Elementor Settings`** > **`Performance`** tab.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂

**Comment by sebastiantheobald** [comment]
Do you will support local FontAwesome Icons? We're struggeling a lot to achieve gdpr compliance and we can not use the default icon library

**Comment by nicholaszein** [comment]
@sebastiantheobald Elementor's FontAwesome implementation is already local. If you're talking about FontAwesome Pro, you can install it via Custom Fonts.

**Comment by nicholaszein** [comment]
@clmnsm .

For legal reasons, we cannot implement such functionality because FontAwesome Pro is a paid service, and this would break their Terms of Service. Their CDN already does what you are suggesting here. We have no plans of adding such functionality as it would break their legal terms.

If your goal is to make their product GDPR compliant (assuming from your statement that it isn't), you should either download and install the icon library locally at your own risk (you can create custom packages), or reach out to them and ask for a GDPR compliant solution.

This is not in our scope.
